### DISCUSSION TITLE
Swagger UI not showing dropdown for Enum request parameter with only one value.

### DISCUSSION THREAD

**Discussion by mausam20** [fastapi/fastapi#11027] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import Enum
class RegionEnum(str, Enum):
    in = "India"
class QueryParams:
    def __init__(
        self,
        region: RegionEnum,
        level: MarketCompanyMarketsLevel):
    self.rerion = region
```

### Description

I am using the above query params in my api and while opening the swagger ui It is showing textbox rather than dropdown for region, 
![Screenshot from 2024-01-25 16-13-18](https://github.com/tiangolo/fastapi/assets/26969620/70ccca33-9384-4d7c-99cd-3f4e6ac90045)
and I need the field to be viewed as
![Screenshot from 2024-01-25 16-14-59](https://github.com/tiangolo/fastapi/assets/26969620/6553b0f0-ab50-43b9-89ee-5a4e8da02ea7)

I have updated the fastapi version to 0.109.0

### Operating System

Linux

### Operating System Details

VERSION="20.04.6 LTS (Focal Fossa)"
VERSION_ID="20.04"
VERSION_CODENAME=focal

### FastAPI Version

0.019.0

### Pydantic Version

2.5.3

### Python Version

3.10

**Comment by JavierSanchezCastro** [comment, marked as answer]
Hi!

I just checked with an own example and doesn't show the dropdown menu, despite that, is it useful to have an Enum with a single value?

**Reply by mausam20** [reply]
Yeah I also think with single value enum is not ideal to use, actually we are migrating out Fastapi application from 0.97.0 to 0.109.0 and the dropdown was coming in previous version for the same,
Thanks @JavierSanchezCastro for your input.

**Reply by mausam20** [reply]
So I have checked this further, due to project requirements we did not want to break the previous UI which was supported with pydantic V1 and fastapi 0.97.0, 
So I checked further that if we put the query in this format then for single enum value and pydantic V2 it is showing the dropdown.
it would be useful to include this example in fastapi documentation as there are nothing similar exist.

```region: RegionEnum = Query(...,enum=[region for region in RegionEnum])```

**Reply by boh5** [reply]
I am currently using fastapi==0.104.1 and pydantic==2.6.0. However, the dropdown menu is not being displayed.💔

**Reply by AyrtonDare** [reply]
I'm experiencing the same with fastapi==0.109.2

**Reply by JavierSanchezCastro** [reply]
Do you have an example?

If you have an enum with a single value (weird case), you can use what @mausam20 said before, something like: 

```python3
from fastapi import FastAPI, Query
from typing import Annotated
from enum import Enum

class RegionEnum(Enum):
    india = "India"

app = FastAPI()

@app.post("/")
def test(q: Annotated[RegionEnum, Query(enum=[region for region in RegionEnum])]=None):
    return None
```

**Reply by AyrtonDare** [reply]
I'm trying with multiple. Using the approach described results in duplicates as both the name and values are shown

```python
from fastapi import FastAPI, Query
from typing import Annotated
from enum import Enum

class RegionEnum(Enum):
    india = "India"
    china = "China"

app = FastAPI()

@app.post("/test")
def test(q: Annotated[RegionEnum, Query(enum=[region for region in RegionEnum])]=None):
    return None
```

<img width="715" alt="image" src="https://github.com/tiangolo/fastapi/assets/133678331/8fd04970-4278-4362-b671-f805f731b43d">

<br>

If I just pass the Enum then it is properly populated, the issue is that I want it to be optional. As soon as I add the optional type-hint and set the default to None the dropdown dissapears.

```python
from fastapi import FastAPI
from typing import Optional
from enum import Enum

class RegionEnum(Enum):
    india = "India"
    china = "China"

app = FastAPI()

@app.post("/test")
def test(q: Optional[RegionEnum] = None):
    return None
```

<img width="702" alt="image" src="https://github.com/tiangolo/fastapi/assets/133678331/d3ab9939-ac02-4776-858d-32dc0f31cf57">

<br>

I've got a workaround but the behaviour I expected was that it would provide the dropdown with just `Optional[RegionEnum] = None`

```python
from fastapi import FastAPI, Query
from typing import Optional
from enum import Enum

class RegionEnum(Enum):
    india = "India"
    china = "China"

regions = {
    region.name: region.value 
    for region 
    in RegionEnum
}

app = FastAPI()

@app.post("/test")
def test(q: Optional[str] = Query(default=None, enum=list(regions.keys()))):
    if q is not None:
        q = regions[q]
    return None
```

<img width="392" alt="image" src="https://github.com/tiangolo/fastapi/assets/133678331/7393c027-b208-415b-80aa-52d42cce97de">

**Reply by JavierSanchezCastro** [reply]
Well for the first example makes no sense to do the last solution since you have more than one item in the enum, and it's for that you're getting duplicate entries.

For the second, your problem is not about enum itself, it's a known issue using Optional. Instead, use `SkipJsonSchema` like:

```python3
from fastapi import FastAPI
from typing import Optional
from enum import Enum
from pydantic.json_schema import SkipJsonSchema

class RegionEnum(Enum):
    india = "India"
    china = "China"

app = FastAPI()

@app.post("/test")
def test(q: RegionEnum | SkipJsonSchema[None] = None):
    return None
```

It's easier to read than the other workaround.

**Reply by AyrtonDare** [reply]
This is perfect, thank you @JavierSanchezCastro
